Code/Framework/AzToolsFramework/AzToolsFramework/UI/PropertyEditor/PropertyFilePathCtrl.cpp

```diff
@@ -26,22 +26,21 @@ AZ_POP_DISABLE_WARNING
 
 namespace AzToolsFramework
 {
-    AZ::IO::Path GetAbsolutePathFromRelativePath(const AZ::IO::Path& relativePath, const AZStd::string& productExtension)
+    AZ::IO::Path GetAbsolutePathFromRelativePath(const AZ::IO::Path& relativePath)
     {
-        // The GetFullSourcePathFromRelativeProductPath API only works on product paths,
-        // so we need to append the product extension to try and find it
-        AZStd::string productRelativePath = relativePath.Native() + productExtension;
-
-        AZStd::string fullPath;
+        // First check if this source relative path already exists and is in an asset safe folder
+        AZ::Data::AssetInfo info;
+        AZStd::string watchFolder;
         bool fullPathIsValid = false;
-        AssetSystemRequestBus::BroadcastResult(
-            fullPathIsValid, &AssetSystemRequestBus::Events::GetFullSourcePathFromRelativeProductPath, productRelativePath, fullPath);
+        AzToolsFramework::AssetSystemRequestBus::BroadcastResult(
+            fullPathIsValid, &AzToolsFramework::AssetSystemRequestBus::Events::GetSourceInfoBySourcePath, relativePath.c_str(), info,
+            watchFolder);
 
         // The full source path asset exists on disk, so use that as
         // the pre-selected file when the user opens the file picker dialog
         if (fullPathIsValid)
         {
-            return fullPath;
+            return AZ::IO::Path(watchFolder) / relativePath;
```

Comment:
This should be using the relative path inside the asset info structure

Reply:
The relative path inside the asset info structure is a string but we want an `AZ::IO::Path` which relativePath is already that type and will match since that's what we supplied to the API call.